### PR TITLE
POST `create-proposal` router created

### PR DESCRIPTION
**Pull request by Muhammed-Rahif**
## Tasks :- 
- [x] Create proposals model
- [x] Create POST router `/create-proposal`
- [x] Validate proposal fields
- [x] Save to database
- [x] Send response to client

### CODE REVIEW

**Comment by Muhammed-Rahif** [comment]
@shamilkotta ready for review!

File: helpers/developerHelpers.js [hunk 1]
```diff
@@ -1,5 +1,13 @@
...
-   },
+   createProposal: proposalData =>
+      new Promise((resolve, reject) => {
+         new Promise((resolve, reject) => {
+            const proposal = new ProposalModel(proposalData);
+            proposal.save((err, newProposal) => {
+               err ? reject(err) : resolve(newProposal);
+            });
+         });
+      }),
```
Comment (shamilkotta) [inline helpers/developerHelpers.js:12]:
```suggestion
      new Promise((resolve, reject) => {
          const proposal = new ProposalModel(proposalData);
          proposal.save((err, newProposal) => {
             err ? reject(err) : resolve(newProposal);
          });
      }),
```

Why nested Promises ?

Comment (Muhammed-Rahif) [reply helpers/developerHelpers.js]: @shamilkotta It happened by mistake when copying and pasting from somewhere else.!

**Review by shamilkotta — APPROVED** [review]
👍

**Pull request merged**